internal/transport/http2_server.go

```diff
@@ -485,14 +488,10 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 	} else {
 		s.ctx, s.cancel = context.WithCancel(t.ctx)
 	}
-	pr := &peer.Peer{
-		Addr: t.remoteAddr,
-	}
-	// Attach Auth info if there is any.
-	if t.authInfo != nil {
-		pr.AuthInfo = t.authInfo
-	}
-	s.ctx = peer.NewContext(s.ctx, pr)
+
+	// Add peer information to the stream context.
+	s.ctx = peer.NewContext(s.ctx, t.getPeer())
```

Comment:
Since this is derived from the transport's context (immediately above), we shouldn't need to re-attach the peer here.